pkg/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.targets

```diff
@@ -1,7 +1,203 @@
-<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+<Project 
+  xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
-    <_ILCompilerRootDir>$(MSBuildThisFileDirectory)../</_ILCompilerRootDir>
-    <IlCompilerTargetsDir>$(_ILCompilerRootDir)targets/</IlCompilerTargetsDir>
+    <DeployAppHost>false</DeployAppHost>
+    <_ILCompilerPackageRootDir>$(MSBuildThisFileDirectory)../</_ILCompilerPackageRootDir>
+    <IlCompilerTargetsDir>$(_ILCompilerPackageRootDir)targets/</IlCompilerTargetsDir>
+   
+    <!-- Workaround for lack of current host OS detection - https://github.com/Microsoft/msbuild/issues/539 -->
+    <TargetOS Condition="'$(TargetOS)' == '' and '$(OS)' == 'Unix' and Exists('/Applications')">OSX</TargetOS>
+    <!-- The correct OS detection code. Uncomment once CI machines are upgraded to the latest version of MSBuild -->
+    <!-- <TargetOS Condition="'$([MSBuild]::IsOSPlatform(OSX))' == 'true'">OSX</TargetOS>  -->
+    <TargetOS Condition="'$(TargetOS)' == ''">$(OS)</TargetOS>
+    
+    <TargetRuntime Condition="'$(TargetOS)' == 'Windows_NT'">win-x64</TargetRuntime>
+    <TargetRuntime Condition="'$(TargetOS)' == 'Linux'">linux-x64</TargetRuntime>
+    <TargetRuntime Condition="'$(TargetOS)' == 'OSX'">osx-x64</TargetRuntime>
+    <_RuntimeILCompilerPackageName Condition="'$(TargetRuntime)' != ''">runtime.$(TargetRuntime).Microsoft.DotNet.ILCompiler</_RuntimeILCompilerPackageName>
+    <RuntimeILCompilerBuildTargetPath>$(IlCompilerTargetsDir)$(_RuntimeILCompilerPackageName).targets</RuntimeILCompilerBuildTargetPath>
   </PropertyGroup>
-  <Import Project="$(IlCompilerTargetsDir)/Microsoft.NETCore.Native.targets"/>
+
+  <!-- Everything below is a temporary hack to enable runtime package resolution. Should be deleted once MSBuild in .NET Core allows importing of build artifacts 
```

Comment:
Is there MSBuild issue that tracks this?

Reply:
Added : https://github.com/Microsoft/msbuild/issues/2807 
Hopefully they can add some clarity to this.